practice_app/api/views.py

```diff
@@ -421,3 +421,41 @@ def user_registration(request):
     else:
         return JsonResponse({"status":"Username is already taken."}, status = 409)
 
+
+@csrf_exempt
+def save_paper_list(request):
+
+    user = request.user
+
+    if user.is_anonymous:
+        # If user us anonymous and credentials are not provided in headers return an error
+        if 'username' not in request.headers or 'password' not in request.headers:
+            return JsonResponse({'status': 'Empty username or password!'}, status=407)
+        # If there are credential information in headers authenticate the user
+        username = request.headers['username']
+        password = request.headers['password']
+        user = authenticate(request, username=username, password=password)
+        if user == None:
+            # If the authentication is failed return an error
+            return JsonResponse({'status' : 'Incorrect username or password!'},status=401)
+
+    # Get the paper list id with the POST method    
+    post_id = request.POST['paper_list_id']
+
+    #Check if the provided id is valid
+    if not models.PaperList.objects.filter(id = post_id).exists():
+        return JsonResponse({'status': 'Paper list is not found!'}, status=404)
+    else:
+        # Get the paper list object and add the current logged in user to the savers list of the paper list
+        paper_list = models.PaperList.objects.get(pk = post_id)
+        username = request.headers['username']
+        password = request.headers['password']
```

Comment:
Are you using these username and password variables anywhere? Please delete these lines, if they are redundant.

Reply:
I used them in the next line to authenticate the user.

Reply:
I believe you're talking about the line 435 and 436

Reply:
Oh, I noticed now. I removed those redundant lines.